addons/account/static/src/js/tax_group.js

```diff
@@ -0,0 +1,269 @@
+         * 
+         * @param {*} record 
+         * @param {*} event 
+         */
+        _reset: function (record, event) {
+            this._super.apply(this, arguments);
+            var self = this;
+            if (event.data.changes.invoice_line_ids) {
+                self.trigger_up('field_changed', {
+                    dataPointID: self.record.id,
+                    changes: { line_ids: record.data.line_ids },
+                });
+            }
+        },
+
+        /**
+         * This methode is responsible for modifying the value of the tax
+         * group by the new value entered by the user. It is also
+         * responsible for modifying taxes accordingly.
+         * 
+         * @param {event} ev 
+         */
+        _setTaxGroups: function (ev) {
+            var myInput;
+            if (ev.target.nodeName === 'INPUT') {
+                myInput = ev.target;
+            } else {
+                return;
+            }
+            // Get the tax group id.
+            var taxGroupId = $(myInput.parentNode.parentNode).attr('data-tax-group-id');
+            // Get the new value of tax group and parse it.
+            var newValue = $(myInput).val();
+            if (!newValue || isNaN(newValue)){
+                var valueBeforeChange = this.record.data.amount_by_group.filter(elem => elem[6] == taxGroupId)[0][8];
+                $(myInput).val(valueBeforeChange);
+                $(myInput).select();
+                return;
+            }
+            newValue = fieldUtils.parse.float(newValue);
+            var newValueFormated = fieldUtils.format.monetary(newValue, {}, { 'currency_id': this.record.data.currency_id.res_id });
+            // Replace the old values.
+            this.record.data.amount_by_group.filter(elem => elem[6] == taxGroupId).forEach(elem => {
+                elem[1] = newValue;
+                elem[3] = newValueFormated;
+            });
+            $(myInput.parentNode.parentNode.children[0]).show(); // Show the pencil.
+            $(myInput.parentNode.parentNode.children[1]).html(newValueFormated); // Write the new value in the HTML.
+            // Change tax value according to the new tax group value.
+            this._changeTaxValueByTaxGroup(this.record.data.amount_by_group.filter(elem => elem[6] == taxGroupId)[0]);
+        },
+
+        //--------------------------------------------------------------------------
+        // Handler
+        //--------------------------------------------------------------------------
+
+        /**
+         * This method is called when the user is in edit mode and 
+         * leaves the <input> field. Then, we execute the code that 
+         * modifies the information.
+         * 
+         * @param {event} ev 
+         */
+        _onBlur: function (ev) {
+            ev.preventDefault();
+            if (this.mode !== 'edit' || !this._isPurchaseDocument()) { // Trigger only in edit mode.
```

Comment:
You could omit this check as the only way to trigger this event is if we are already in edit mode and in a purchase document